### PR TITLE
Res 484 add pose refinement methods

### PR DESCRIPTION
**Pull request by homee-james**
Summary:

Add three pose refinement methods.

glomap (global sfm method. [ref](https://github.com/colmap/glomap))
loftr (dense matching method which performs better on texture-less scenes. [ref](https://github.com/zju3dv/LoFTR))
lightglue (faster version of superglue. [ref](https://github.com/cvg/LightGlue))
Details:

Each method can be found in arkit_utils/pose_optimization.
To call the method, uncomment the targeted method and comment the rest methods at fourth step in run_arkit_3dgs.sh.

### CODE REVIEW

File: arkit_utils/pose_optimization/optimize_pose_loftr.py
Comment (homee-dennis) [inline arkit_utils/pose_optimization/optimize_pose_loftr.py:1]: I remember that when first read_model in .bin format (line132) and write_model in .txt format (line 133) will cause the miss ordering in poses. If I’m mistaken, please correct me.

File: scene/gaussian_mesh_model.py
Comment (homee-dennis) [inline scene/gaussian_mesh_model.py:1]: nice correction on a1,a2,a3 need to both greater than 0 not sum :)

File: train.py
Comment (homee-dennis) [inline train.py:1]: I like the coding style you re-org the params in different lines 👍

**Review by homee-dennis — COMMENTED** [review]
LGTM, the only concern is the read write model part of code.

File: arkit_utils/pose_optimization/optimize_pose_loftr.py
Comment (homee-james) [reply arkit_utils/pose_optimization/optimize_pose_loftr.py:1]: fix at [6b9eee6](https://github.com/homee-ai/gaussian-splatting/pull/4/commits/6b9eee67aa4a19c8f4b0f3df93e3a8232b11ef7b) to sort the model ascendly before writing.

**Comment by homee-james** [comment]
Add a few modification:
- Sort the colmap model before writing [6b9eee6](https://github.com/homee-ai/gaussian-splatting/pull/4/commits/6b9eee67aa4a19c8f4b0f3df93e3a8232b11ef7b)
- Refactor preprocessing code( add prepare_dataset.sh and deprecate run_arkit_3dgs.sh) [8212952](https://github.com/homee-ai/gaussian-splatting/pull/4/commits/82129527c4807185690e0d3311cd4cef725c9e69)
- Add colored logging due to complicated logging [78ee77b](https://github.com/homee-ai/gaussian-splatting/pull/4/commits/78ee77b022e5ccf36a497dffd3cb3da7fd422edf)
- Update README.MD for preprocessing tutorial and output format [8212952](https://github.com/homee-ai/gaussian-splatting/pull/4/commits/82129527c4807185690e0d3311cd4cef725c9e69)

**Comment by homee-dennis** [comment]
LGTM, the data structure and pose optimization pipeline are more compact and organized.
Feel free to merge.

**Pull request merged**